src/zkbob/utils/PriorityQueue.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.13;
+
+/**
+ * @dev The structure that stores all information about the pending commitment.
+ * @param commitment commitment value to be added in the Merkle Tree.
+ * @param prover address of the prover that submitted the commitment.
+ * @param fee fee reserved for the prover who will submit the tree update proof.
+ * @param timestamp commitment timestamp.
+ */
+struct PendingCommitment {
+    uint256 commitment;
+    address prover;
+    uint64 fee;
+    uint64 timestamp;
```

Comment:
I think it's worth packing timestamp in `uint32`, so that all `prover`, `fee` and `timestamp` sit in a single storage slot.

Reply:
Yes, thank you. I didn't think about it. Hope users from 2106 will not hate us :)

Reply:
Fixed in https://github.com/zkBob/zkbob-contracts/pull/97/commits/6d16f95b0ac5a7bbefdf566e022423d8e7062308